cmds/core/uptime/uptime.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Get the time the machine has been up
+//
+// Synopsis:
+//     uptime
+//
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// This function takes in the filepath to a file containing loadavgs,it then extracts and returns the three load averages
+func loadavg(filepath string) (load1, load2, load3 string, err error) {
+	loadaverages, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return "-1", "-1", "-1", errors.New("Invalid File Contents")
+	}
+	loadAvgArray := strings.Split(string(loadaverages), " ")
+	return loadAvgArray[0], loadAvgArray[1], loadAvgArray[2], nil
+}
+
+// This function takes in the filepath to a file containing the uptime,it then extracts and returns the uptime in the formart
+// Days , Hours , Minutes ,Seconds
+func uptime(filepath string) (days, hours, minutes int, err error) {
+
+	uptime, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return -1, -1, -1, errors.New("Invalid File")
+	}
+	uptimeString := string(uptime)
+	uptimeArray := strings.Split(uptimeString, " ")
+	uptimeSeconds := strings.Split(uptimeArray[0], ".")
+	uptimeInteger, err := strconv.Atoi(uptimeSeconds[0])
```

Comment:
I agree ,this is a clever way of avoiding all the errors that come with calculations and using the atoi function.I re-implemented the uptime function using your suggestion and it worked perfectly.I had to subtract 1 hour from the t.Hours() because due to relative time of zero , t.Hours was always 1 more than the actual correct number of hours.Done.